### DISCUSSION TITLE
how to apply drop_edge function to a SparseTensor.

### DISCUSSION THREAD

**Discussion by kou18n** [pyg-team/pytorch_geometric#7562] [Q&A]
If the edge_index is a SparseTensor. How to apply the drop_edge function?
```
SparseTensor(row=tensor([     0,      0,      1,  ..., 373171, 373172, 373172], device='cuda:0'),
             col=tensor([     0,      2,      0,  ..., 373171, 373169, 373172], device='cuda:0'),
             size=(373173, 373173), nnz=1113071, density=0.00%)
```

**Comment by wsad1** [comment]
At the moment [dropout_edge](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/utils/dropout.html#dropout_edge) doesn't support `SparseTensor` as input, but that should be easy to support, will send a PR for that.
For now you can do that following. 
```
from torch_geometric.utils import dropout_edge, to_edge_index
sparse_edge_index = SparseTensor(...)
edge_index, _ = to_edge_index(sparse_edge_index)
edge_index, _ = dropout_edge(edge_index, p=0.5)
sparse_edge_index = SparseTensor(row = edge_index[0], col = edge_index[1], sparse_sizes=(3, 3))
```

**Reply by kou18n** [reply]
> At the moment [dropout_edge](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/utils/dropout.html#dropout_edge) doesn't support `SparseTensor` as input, but that should be easy to support, will send a PR for that. For now you can do that following.
> 
> ```
> from torch_geometric.utils import dropout_edge, to_edge_index
> sparse_edge_index = SparseTensor(...)
> edge_index, _ = to_edge_index(sparse_edge_index)
> edge_index, _ = dropout_edge(edge_index, p=0.5)
> sparse_edge_index = SparseTensor(row = edge_index[0], col = edge_index[1], sparse_sizes=(3, 3))
> ```
Thank you for your quickly reply! I will try your code!

**Comment by kou18n** [comment]
Is there any way to set the seed, and can reproduce the result of the dropout_edge function result?

**Reply by EdisonLeeeee** [reply]
```python
from torch_geometric import seed_everything

seed_everything(your_seed)
```

**Reply by kou18n** [reply]
Thank you for reply. After dropout_node or dropout_edge function, I think the original graph will change, and the `graph.x` node features will change, how should I update it?

**Reply by rusty1s** [reply]
I don't see a reason why you would want to change the graph size after applying `dropout_edge`. For changing the graph size in `dropout_node`, you can do
```python
data.edge_index, edge_mask, node_mask = dropout_node(...)
data.x = data.x[node_mask]

**Reply by kou18n** [reply]
Thank you for reply.
I'm planning to implement the GraphCL model, and I need to generate two graph augmentation `Aug1` and `Aug2`. 
So I want to use `dropout_node` function to get one of the graph augmentation 'Aug1'. I think Aug1's node feature dim is differet with the orginal graph, maybe need to remove the dropped node feature from the orignal graph to get a new aug1 graph.

Original graph:
`Data(y=[1], x=[1611, 5], adj_t=[1611, 1611, nnz=8759], num_nodes=1611)`
Through `dropout_node`  for example, 100 nodes have dropped.
Aug1 will become :
`Data(y=[1], x=[1511, 5], adj_t=[.......], num_nodes=1511)`
I'm not sure if my idea is correct.

**Reply by rusty1s** [reply]
I don't know the GraphCL paper, but what is the problem with that? This looks correct, right?

**Reply by kou18n** [reply]
Thank you for your today‘s answers.
#7583 problem solved.
About #1773, everything is done.
Currently, the code only support spares_tensor.

**Reply by rusty1s** [reply]
Will take a look, thanks!
